app/controllers/publish_interface/providers/contacts_controller.rb

```diff
@@ -0,0 +1,40 @@
+module PublishInterface
+  module Providers
+    class ContactsController < PublishInterfaceController
+      def edit
+        @provider_contact_form = ProviderContactForm.new(provider)
+      end
+
+      def update
+        @provider_contact_form = ProviderContactForm.new(provider, params: provider_contact_params)
+
+        if @provider_contact_form.save!
+          flash[:success] = I18n.t("success.published")
+
+          redirect_to details_publish_provider_recruitment_cycle_path(
+            provider.provider_code,
+            recruitment_cycle.year,
+          )
+        else
+          render :edit
+        end
+      end
```

Comment:
Both of these actions need to authorize access, like so https://github.com/DFE-Digital/teacher-training-api/pull/2358/files#diff-c8f9f288e415f332553b657513de9c2842f029c15424dac1462667cf024a26ddR22

Reply:
Come to think of it, the Visas controller needs to do the same

Reply:
done